pkg/ccl/importccl/read_import_mysql.go

```diff
@@ -0,0 +1,144 @@
+// Copyright 2018 The Cockroach Authors.
+//
+// Licensed as a CockroachDB Enterprise file under the Cockroach Community
+// License (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+//     https://github.com/cockroachdb/cockroach/blob/master/licenses/CCL.txt
+
+package importccl
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/pkg/errors"
+	mysql "github.com/xwb1989/sqlparser"
+
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/types"
+	"github.com/cockroachdb/cockroach/pkg/sql/sessiondata"
+	"github.com/cockroachdb/cockroach/pkg/sql/sqlbase"
+	"github.com/cockroachdb/cockroach/pkg/util/ctxgroup"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
+)
+
+type mysqldumpReader struct {
+	conv     rowConverter
+	debugRow func(tree.Datums)
+}
+
+var _ inputConverter = &mysqldumpReader{}
+
+func newMysqldumpReader(
+	kvCh chan kvBatch, table *sqlbase.TableDescriptor,
+) (*mysqldumpReader, error) {
+	conv, err := newRowConverter(table, kvCh)
+	if err != nil {
+		return nil, err
+	}
+	return &mysqldumpReader{conv: *conv}, err
+}
+
+func (m *mysqldumpReader) start(ctx ctxgroup.Group) {
+}
+
+func (m *mysqldumpReader) inputFinished(ctx context.Context) {
+	close(m.conv.kvCh)
```

Comment:
since all of these have to do this could we make the caller of inputFinished close the chan instead? don't care either way.

Reply:
well the ones that use separate convertCh close their internal converter ch, then when the workers finish draining *they* close kvCh.

Reply:
right. ok.